### ISSUE TITLE
[Bug] Incorrect Selector & String Parsing

### ISSUE THREAD

**Issue by MerlinofMines** [MerlinofMines/EasyCommands#233]
**Describe the bug**
There are a couple parsing issues when trying to set selector values to a string when the string being set includes a block type keyword.

**To Reproduce**
```
set myList to []
set myList["thing"] to "My Drill"
```

```
set "LCD Panel" display text to "Drill : a"
```

**Expected behavior**
For the first, "My Drill" should be added to the list.
For the second, the "LCD Panel" display text should be set to "Drill : a"

**Comment by jgersti** [comment]
> ```
> set "LCD Panel" display text to "Drill : a"
> ```

The corresponding parse is
```
    4: [Assignment] [AmbiguousString] [BlockType] [Property] [Ignore] *[AmbiguousString]
    4: [Assignment] {*[AmbiguousString] [BlockType]} [Property] [Ignore] [Selector]
   18: [Assignment] [Selector] [Property] *[Ignore] [Selector]
   20: [Assignment] [Selector] *[Property] [Selector]
   61: [Assignment] [Selector] ([PropertySupplier] *[Selector])
   61: ([Assignment] *[Selector] [Variable])
       : [Command]
```

**Comment by jgersti** [comment]
> ```
> set myList to []
> set myList["thing"] to "My Drill"
> ```

```
    1: [Assignment] [AmbiguousString] [Ignore] [OpenBracket] [CloseBracket]
    4: [Assignment] [AmbiguousString] [Ignore] [List]
   17: [Assignment] [Variable] [Ignore] [List]
   18: [Assignment] [Variable] [Ignore] [ListIndex]
   23: [Assignment] [Variable] [ListIndex]
   29: [VariableAssignment] [ListIndex]
   67: [VariableAssignment] [Variable]
     : [Command]
```

**Comment by jgersti** [comment]
### TL:DR
rule 4.1 needs to have lower priority in the branching since it should only be used for convenience anyway (`$` is there for a reason)

**Comment by MerlinofMines** [comment]
This was resolved by #234
